provisioning/transport/http/src/http_transport.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { EventEmitter } from 'events';
+import { X509 } from 'azure-iot-common';
+import { ProvisioningTransportOptions, ProvisioningTransportHandlersX509, PollingStateMachine } from 'azure-iot-provisioning-device';
+import { Http as Base } from 'azure-iot-http-base';
+import { HttpPollingHandlers } from './http_polling_handlers';
+
+export class Http  extends EventEmitter implements ProvisioningTransportHandlersX509 {
+  private _stateMachine: PollingStateMachine;
+  private _pollingHandlers: HttpPollingHandlers;
+
+  constructor(idScope: number, httpBase?: Base) {
+    super();
+    this._pollingHandlers = new HttpPollingHandlers(idScope, httpBase);
+    this._stateMachine = new PollingStateMachine(this._pollingHandlers);
+
+    this._stateMachine.on('operationStatus', (eventBody) => {
+      this.emit('operationStatus', eventBody);
+    });
+  }
+
+  endSession(callback: (err?: Error) => void): void {
+    this._stateMachine.endSession(callback);
+  }
+
+  setTransportOptions(options: ProvisioningTransportOptions): void {
+    this._pollingHandlers.setTransportOptions(options);
+  }
+
+  registerX509(registrationId: string, authorization: X509, forceRegistration: boolean, callback: (err?: Error, assignedHub?: string, deviceId?: string, responseBody?: any, result?: any) => void): void {
```

Comment:
>assignedHub?: string, deviceId?: string [](start = 112, length = 39)

we're going to find these 2 together pretty much all the time, i'd be in favor of creating a RegistrationResult object.

Reply:
Good idea.  The ProvisioningDeviceConfiguration object in interfaces.ts has this.  I'll rename that and use it whenever we have these 2 together

---
In reply to: [151319471](https://github.com/Azure/azure-iot-sdk-node/pull/154#discussion_r151319471) [](ancestors = 151319471)